clients/src/test/java/org/apache/kafka/common/telemetry/internals/MetricNamingConventionTest.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.telemetry.internals;
+
+import org.apache.kafka.common.MetricName;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Collections;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class MetricNamingConventionTest {
+
+    private MetricNamingStrategy<MetricName> metricNamingStrategy;
+
+    @BeforeEach
+    public void setUp() {
+        metricNamingStrategy = MetricNamingConvention
+            .getClientTelemetryMetricNamingStrategy("org.apache.kafka");
+    }
+
+    @Test
+    public void testMetricKey() {
+        MetricName metricName = new MetricName("name", "group", "description",
+            Collections.emptyMap());
+        MetricKey metricKey = metricNamingStrategy.metricKey(metricName);
+
+        assertEquals("org.apache.kafka.group.name", metricKey.getName());
+        assertEquals(Collections.emptyMap(), metricKey.tags());
+    }
+
+    @Test
+    public void testMetricKeyWithHyphenNameAndTags() {
+        Map<String, String> tags = new HashMap<>();
```

Comment:
Based on the test name, we should use `emptyMap` for tags in this test.

Reply:
Corrected test name.